kernel/src/transaction/mod.rs

```diff
@@ -718,97 +1004,204 @@ impl Transaction {
+                let with_new_dv_transform = Expression::transform(
+                    Transform::new_top_level()
+                        .with_replaced_field(
+                            "deletionVector",
+                            Expression::column([NEW_DELETION_VECTOR_NAME]).into(),
+                        )
+                        .with_dropped_field(NEW_DELETION_VECTOR_NAME),
+                );
+                let with_new_dv_eval = evaluation_handler.new_expression_evaluator(
+                    intermediate_dv_schema().clone(),
+                    Arc::new(with_new_dv_transform),
+                    nullable_scan_rows_schema().clone().into(),
+                );
+
+                let with_new_dv_data = with_new_dv_eval.evaluate(file_metadata_batch.data())?;
+
+                let restored_add_eval = evaluation_handler.new_expression_evaluator(
+                    nullable_scan_rows_schema().clone(),
+                    get_scan_metadata_transform_expr(),
+                    nullable_restored_add_schema().clone().into(),
+                );
+                let as_partial_add_data = restored_add_eval.evaluate(with_new_dv_data.as_ref())?;
+
+                let with_data_change_transform =
+                    Expression::transform(Transform::new_nested(["add"]).with_inserted_field(
+                        Some("modificationTime"),
+                        Expression::literal(self.data_change).into(),
+                    ));
+                let expr = Expression::struct_from([with_data_change_transform]);
+                let with_data_change_eval = evaluation_handler.new_expression_evaluator(
+                    nullable_restored_add_schema().clone(),
+                    Arc::new(expr),
+                    nullable_restored_add_with_data_change_schema().into(),
+                );
+                let with_data_change_data =
+                    with_data_change_eval.evaluate(as_partial_add_data.as_ref())?;
+
+                FilteredEngineData::try_new(
+                    with_data_change_data,
+                    file_metadata_batch.selection_vector().to_vec(),
+                )
+            },
+        ))
+    }
+}
+
+struct DvMatchVisitor<'a> {
+    dv_updates: &'a HashMap<String, DeletionVectorDescriptor>,
+    new_dv_entries: Vec<Scalar>,
+    matched_file_indexes: Vec<usize>,
+}
+
+impl<'a> DvMatchVisitor<'a> {
+    const PATH_INDEX: usize = 0;
+
+    fn new(dv_updates: &'a HashMap<String, DeletionVectorDescriptor>) -> Self {
+        Self {
+            dv_updates,
+            new_dv_entries: Vec::new(),
+            matched_file_indexes: Vec::new(),
+        }
+    }
+}
+
+impl RowVisitor for DvMatchVisitor<'_> {
+    fn selected_column_names_and_types(&self) -> (&'static [ColumnName], &'static [DataType]) {
+        static NAMES_AND_TYPES: LazyLock<(Vec<ColumnName>, Vec<DataType>)> = LazyLock::new(|| {
+            let names = vec![column_name!("path")];
```

Comment:
is there an existing constant that we can use here instead of hard-coding the string?